src/content/app/genome-browser/hooks/useGenomicTracks.ts

```diff
@@ -66,6 +67,11 @@ const useGenomicTracks = () => {
     );
     trackIdsList.forEach(genomeBrowserMethods.toggleTrack);
     genomeIdInitialisedRef.current = activeGenomeId as string;
+
+    // reset this ref on unmount to rerun this useEffect on mount for the last selected genome id
```

Comment:
We should probably update — or maybe even delete — this comment. `useEffect`'s cleanup function doesn't run only when the component unmounts, but also when any of the members of the dependences array changes. Since this hook is run in the `Browser` component, we don't expect any unmounting when you switch between species, even if one of the species shows the interstitial. The comment doesn't really add anything to what's already written on line 73.

Reply:
That's done now ✅